src/mono/netcore/Makefile

```diff
@@ -0,0 +1,57 @@
+DOTNET_DIR=../../../.dotnet
+DOTNET=$(DOTNET_DIR)/dotnet
+TESTHOST_PATH=../../../artifacts/bin/testhost/
+BOOTSTRAP_RUNTIME = $(shell ls $(DOTNET_DIR)/shared/Microsoft.NETCore.App | tail -1)
+
+TARGET_FRAMEWORK=
+CONFIGURATION=
+OS=
+PLATFORM=
+
+# let's check ../../../artifacts/bin/testhost/ for available configurations
+# it's expected to see there only a single config, e.g. "netcoreapp5.0-OSX-Release-x64"
+# otherwise you have to set these variables for desired config
+ifeq ($(words $(wildcard $(TESTHOST_PATH)/*)), 1)
+	TESTHOST_CFG := $(subst -, ,$(notdir $(wildcard $(TESTHOST_PATH)/*)))
+	TARGET_FRAMEWORK := $(word 1,$(TESTHOST_CFG))
+	OS := $(word 2,$(TESTHOST_CFG))
+	CONFIGURATION := $(word 3,$(TESTHOST_CFG))
+	PLATFORM := $(word 4,$(TESTHOST_CFG))
+endif
+
+ifeq ($(OS), OSX)
+	NATIVE_EXT := .dylib
+else
+	NATIVE_EXT := .so
+endif
+
+check-config:
+	@if test -z "$(CONFIGURATION)"; then echo "CONFIGURATION is not set"; exit 1; fi
+
+run-sample: check-config
+	$(DOTNET) msbuild /t:RunSample /p:Configuration=$(CONFIGURATION) ../mono.proj
+
+runtime:
+	$(DOTNET) msbuild /t:Build ../mono.proj
+
+# temp: makes $(DOTNET) to use mono runtime (to run real-world apps using '$(DOTNET) run')
+patch-local-dotnet: check-config
+	cp ../../../artifacts/obj/mono/$(OS).$(PLATFORM).$(CONFIGURATION)/mono/mini/.libs/libmonosgen-2.0$(NATIVE_EXT) $(DOTNET_DIR)/shared/Microsoft.NETCore.App/$(BOOTSTRAP_RUNTIME)/libcoreclr$(NATIVE_EXT)
```

Comment:
this can copy from `../../../artifacts/bin/mono/$(OS).$(PLATFORM).$(CONFIGURATION)/libcoreclr.dylib` since that one is already the renamed libmonosgen.

Reply:
but I think all of this logic should be in the mono.proj instead. the only purpose of this makefile should be running appropriate msbuild targets, but no actual logic.